src/lib/storage/zero_suppression/simd_bp128/simd_bp128_vector.cpp

```diff
@@ -0,0 +1,30 @@
+#include "simd_bp128_vector.hpp"
+
+namespace opossum {
+
+SimdBp128Vector::SimdBp128Vector(pmr_vector<uint128_t> vector, size_t size) : _data{std::move(vector)}, _size{size} {}
+
+const pmr_vector<uint128_t>& SimdBp128Vector::data() const { return _data; }
+
+size_t SimdBp128Vector::_on_size() const { return _size; }
+size_t SimdBp128Vector::_on_data_size() const { return sizeof(uint128_t) * _data.size(); }
+
+std::unique_ptr<BaseZeroSuppressionDecoder> SimdBp128Vector::_on_create_base_decoder() const {
+  return std::unique_ptr<BaseZeroSuppressionDecoder>{_on_create_decoder()};
+}
+
+std::unique_ptr<SimdBp128Decoder> SimdBp128Vector::_on_create_decoder() const {
+  return std::make_unique<SimdBp128Decoder>(*this);
+}
+
+auto SimdBp128Vector::_on_cbegin() const -> ConstIterator { return ConstIterator{&_data, _size, 0u}; }
```

Comment:
Any reason to explicitly give the return type here? We don't do that anywhere else.

Reply:
Updated.